components/brave_ads/content/browser/creatives/search_result_ad/creative_search_result_ad_mojom_web_page_entities_extractor.cc

```diff
@@ -142,8 +143,17 @@ bool ExtractCreativeAdMojomProperty(
 
   const std::string& property_name = mojom_property->name;
   if (property_name == kCreativeAdPlacementIdPropertyName) {
-    return GetRequiredStringValue(mojom_property,
-                                  &mojom_creative_ad->placement_id);
+    std::string placement_id;
+    if (!GetRequiredStringValue(mojom_property, &placement_id)) {
+      return false;
+    }
+
+    // Escape all characters except alphanumerics and -._~ to make sure that the
+    // placement id can be safely passed to JS function and can be compared with
+    // an encoded placement id from search result click url.
+    mojom_creative_ad->placement_id =
```

Comment:
Escape characters for placement id.

Reply:
`click url` is known as `landing page URL` for creative search result ads

Reply:
Actually this is not a landing page URL. 
It is an URL which is opened when user clicks a search result ad and this navigation is eventually redirected to a landing page.

Reply:
Ah. Thanks